src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -650,6 +652,9 @@ protected void copyResponseEntity(HttpResponse proxyResponse, HttpServletRespons
             }
           }
         } finally {
+          if(proxyResponse instanceof CloseableHttpResponse) {
```

Comment:
I think this is slightly better placed out in the service() method near after when copyResponseEntity is called (i.e. the caller).

Reply:
It appears that the proxy will now always close the proxyResponse.  I thought it would be done conditionally based on something, like maybe if we get an exception during os.write() ?

Reply:
> I think this is slightly better placed out in the service() method near after when copyResponseEntity is called (i.e. the caller).

The placement was intentionally only in the codepath, that handles chunked encodings. I tried to reproduce the blocking problem with a non-chunked stream and there the normal code path works correctly.  This keeps the impact minimal.

> It appears that the proxy will now always close the proxyResponse. I thought it would be done conditionally based on something, like maybe if we get an exception during os.write() ?

Right - I moved the special casing into a catch block, so that a normal chunked transfer is passed through.